cdap-ui/app/services/data/my-cdap-datasource.js

```diff
@@ -29,7 +29,11 @@ angular.module(PKG.name + '.services')
 
       // FIXME: There is a circular dependency and that is why
       // myAuth.isAuthenticated is not used. There should be a better way to do this.
-      if (window.CDAP_CONFIG.securityEnabled && $rootScope.currentUser && $rootScope.currentUser.token) {
+      if (
```

Comment:
Any chance you can address the `FIXME` above?

Reply:
Fair point but as of now no. Ideally I don't want to spend time on this when we know this is going to go away. I would need to spend sometime in investigating why the cyclic dependency happens and clean it up and at this point I don't want to introduce regressions 😭

Reply:
That's fine. I was really wondering if the new changes made it moot.

Reply:
No. The dependency still exists.